### PR TITLE
Simplify dog_tag_dog variants, update snippet documentation

### LINKED ISSUE #68831: Allow item descriptions to reference snippets, the way dialogue does.
### Is your feature request related to a problem? Please describe.

I've been trying to create a few items here recently, and I noticed that I cannot call up snippets for use in their descriptions like I can in dialogue, such as using <given_name> or something similar. This makes item descriptions very limited in what they can say, especially for story-related items like survivor notes or lab notes.

### Solution you would like.

I would like to be able to reference snippets in item descriptions the way I can in dialogue, calling them as needed.

### Describe alternatives you have considered.

Manually creating override strings that feature all possible combinations of the snippets I'm trying to use, which is extremely tedious and takes up way too much space.

### Additional context

None at this moment.

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
~~Fix 68831~~
#### Describe the solution
~~use the code, proposed by Ehughsbaird, in https://github.com/CleverRaven/Cataclysm-DDA/issues/68831#issuecomment-1779946754~~
~~Replace #68864 variants with single item and snippet of all dog names from said pr, as an example~~
See https://github.com/CleverRaven/Cataclysm-DDA/pull/68966#issuecomment-1794913430
#### Testing
image
image
~~Surprisingly, the game handle it super nice, similarly to how variant works - the items do not stack, if they have different description~~
#### Additional context
~~maybe add phone number also, in format of `555-<number><number><number><number>`?
also, ď̴̹o̵͓̐g̵̹͌ ̴̹̃ť̷̘a̶͈̋g̷̳͂~~
GIF 28-10-2023 17-15-14

### CODE REVIEW

**Comment by MNG-cataclysm** [comment]
Thank you for doing this!

**Comment by GuardianDll** [comment]
GIF 28-10-2023 18-56-12

Need to resolve the game not saving the snippet in the item, and generating it on demand

**Comment by GuardianDll** [comment]
Since code change doesn't work as expected, i changed the PR to simply improve the dog_tag_dog description and updating the documentation

**Comment by ehughsbaird** [comment]
For code changes, see https://github.com/CleverRaven/Cataclysm-DDA/pull/69149

File: doc/JSON_INFO.md [hunk 1]
```diff
@@ -667,6 +667,230 @@ This section describes each json file and their contents. Each json has their ow
...
+{
+  "type": "snippet",
+  "category": "<music_description>",
+  "text": [ "some <musicgenre>.", "some <musicgenre>. The <musicdesc_part> is <musicdesc_evaluation>." ]
+},
+```
+
+------
+
+Item descriptions also capable to use snippet system, but to use them, `expand_snippets` should be `true`, otherwise the description won't be saved, and will change every time you update the game in any way, either by picking up the item again, reloading the game or, sometimes, by moving your cursor
```
Comment (ehughsbaird) [inline doc/JSON_INFO.md:753]: Snippets simply won't be expanded if it is not set to true.

File: doc/JSON_INFO.md [hunk 2]
```diff
@@ -667,6 +667,230 @@ This section describes each json file and their contents. Each json has their ow
...
+    }
+  ],
+  "material": [ "ceramic" ],
+  "weight": "375 g",
+  "volume": "375 ml"
+}
+```
+
+Using `expand_snippets` required only where snippets are used - if item do not uses snippet, but variant does, then only variant require to have `"expand_snippets":true`
+Additionally, using `expand_snippets` in item definition works as if all variants got `"expand_snippets": true`
```
[the remark this answers is not in the document]
Comment (GuardianDll) [reply doc/JSON_INFO.md:800]: so it will effectively remove all variants that do not have snippets?

[the remark this answers is not in the document]
Comment (ehughsbaird) [reply doc/JSON_INFO.md:800]: The descriptions, yes.

**Pull request merged**